pkg/assisted/infraenv.go

```diff
@@ -827,6 +831,32 @@ func (builder *InfraEnvBuilder) Delete() (*InfraEnvBuilder, error) {
 	return builder, nil
 }
 
+// DeleteAndWait deletes an InfraEnvBuilder and waits until it is removed from the cluster.
+func (builder *InfraEnvBuilder) DeleteAndWait(timeout time.Duration) error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof(`Deleting InfraEnvBuilder %s and 
+	waiting for the defined period until it's removed`,
+		builder.Definition.Name)
+
+	if _, err := builder.Delete(); err != nil {
+		return err
+	}
+
+	// Polls the InfraEnvBuilder every second until it's removed.
+	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		_, err := builder.Get()
+		if err != nil {
+
+			return true, nil
```

Comment:
shouldn't this be "false, err"

Reply:
Yes, correct. I'll change it thanks

Reply:
This should be true. If err != nil, that means an error occurred because the InfraEnv doesn't exist which is what we want in this case

Reply:
right.